packages/hoprd/src/index.ts

```diff
@@ -171,6 +178,28 @@ async function generateNodeOptions(): Promise<HoprOptions> {
     options.dbPath = argv.data
   }
 
+  const protocolConfig = require('../protocol-config.json') as ProtocolConfig
+  for (const environment of protocolConfig.environments) {
+    if (environment.id === argv.environment) {
+      for (const network of protocolConfig.networks) {
+        if (network.id === environment.network_id) {
+          options.environment = {
+            id: environment.id,
+            network,
+            channel_contract_deploy_block: environment.channel_contract_deploy_block,
+            token_contract_address: environment.token_contract_address,
+            channels_contract_address: environment.channels_contract_address
+          }
+        }
+      }
+
+    }
+  }
+
+  if (!options.environment) {
+    throw new Error(`failed to find environment with id ${argv.environment} in protocol-config.json`)
```

Comment:
The reference to the file doesn't make sense, since the error is user-facing and the file itself a build-time consideration. Something like `in the supported protocol configuration` might be less confusing.